weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/sections/ExampleCompareSection/ExampleCompareSectionTable.tsx

```diff
@@ -0,0 +1,1226 @@
+  useExampleCompareData,
+  useFilteredAggregateRows,
+} from './exampleCompareSectionUtil';
+
+const styledDataGridStyleOverrides: SxProps = {
+  '& .MuiDataGrid-row:hover': {
+    backgroundColor: 'white',
+  },
+  '& .MuiDataGrid-row.Mui-selected:hover': {
+    // match the non-hover background color
+    backgroundColor: 'rgba(169, 237, 242, 0.32)',
+  },
+  '& .MuiDataGrid-cell--pinnedLeft': {
+    backgroundColor: 'white',
+    zIndex: '7 !important',
+  },
+  width: '100%',
+};
+
+/**
+ * Types for the comparison table data structure
+ */
+
+// Base type for all row data
+type BaseRowData = Pick<PivotedRow, 'output' | 'scores' | 'inputDigest'> & {
+  id: string;
+  _expansionId: string;
+};
+
+// Row data when models are displayed as columns
+type ModelAsColumnsRowData = BaseRowData & {
+  _pivot: 'modelsAsColumns';
+};
+
+// Row data when models are displayed as rows
+type ModelAsRowsRowData = BaseRowData &
+  Pick<PivotedRow, 'evaluationCallId' | 'predictAndScore'> & {
+    _pivot: 'modelsAsRows';
+  };
+
+type RowDataBase = ModelAsColumnsRowData | ModelAsRowsRowData;
+
+// Row data for individual trials
+type TrialRowData = RowDataBase & {
+  _type: 'trial';
+  _trialNdx: number;
+};
+
+// Row data for summary rows
+type SummaryRowData = RowDataBase & {
+  _type: 'summary';
+  _numTrials: number;
+};
+
+type RowData = TrialRowData | SummaryRowData;
+
+/**
+ * Component Props
+ */
+interface DatasetRowItemRendererProps {
+  digest: string;
+  inputKey: string;
+}
+
+interface ExampleCompareSectionTableProps {
+  state: EvaluationComparisonState;
+  shouldHighlightSelectedRow?: boolean;
+  onShowSplitView: () => void;
+}
+
+/**
+ * Constants
+ */
+const DISABLED_ROW_SPANNING = {
+  rowSpanValueGetter: (value: any, row: RowData) => row.id,
+};
+
+const SCORE_COLUMN_SETTINGS = {
+  flex: 1,
+  minWidth: 120,
+};
+
+// Constants for column width calculations
+const MIN_COLUMN_WIDTH = 100;
+const MAX_COLUMN_WIDTH = 400;
+const DYNAMIC_COLUMN_MAX_WIDTH = 500;
+const BASE_CHAR_WIDTH = 8; // Approximate width of a character in pixels
+const PADDING = 32; // Padding for cell content
+
+// Helper to estimate content width
+const estimateContentWidth = (content: any): number => {
+  if (content == null) return MIN_COLUMN_WIDTH;
+
+  if (typeof content === 'object' && '_type' in content) {
+    return MIN_COLUMN_WIDTH;
+  }
+
+  // Convert content to string for length estimation
+  const contentStr =
+    typeof content === 'string' ? content : JSON.stringify(content);
+
+  // Calculate width based on content length
+  const estimatedWidth = Math.min(
+    Math.max(contentStr.length * BASE_CHAR_WIDTH + PADDING, MIN_COLUMN_WIDTH),
+    MAX_COLUMN_WIDTH
+  );
+
+  return estimatedWidth;
+};
+
+// Hook to calculate column widths based on first row
+const useColumnWidths = (
+  ctx: CompareEvaluationContext,
+  inputSubFields: string[],
+  outputColumnKeys: string[],
+  rows: RowData[]
+) => {
+  const firstRow = useFirstExampleRow(ctx);
```

Comment:
not blocking: I think this heuristic is a reasonable start, but I wonder what happens if the first row is `N/A`, and subsequent rows are much longer...

Reply:
👍 this is going to change in the near future anyhow so leaving for now